### PR TITLE
Feat/pagination

### PR DESCRIPTION
**Pull request by amalasyrafs**

### CODE REVIEW

File: packages/react/src/components/pagination.tsx
Comment (irfancoder) [inline packages/react/src/components/pagination.tsx:1]: 1. Lacking support for customisation of pagination items. Eg. how to handle i18n in `Previous` & `Next`, use different icon etc
2. Can you give an example how to integrate this with server pagination? ie. where the state is taken from query params
3. Rename:
- ~~basic~~ `simple`
- ~~number~~ `default`
- ~~basic-alternate~~ `full`

**Review by irfancoder — CHANGES_REQUESTED** [review]

**Comment by amalasyrafs** [comment]
hi @irfancoder ,

rename the types for pagination and added server example. but i have to add new page `pagination-server.stories.tsx` to be able to exhibit this example. 

or if you know any other method to do this, please lmk.

File: apps/storybook/stories/pagination-server.stories.tsx
Comment (irfancoder) [inline apps/storybook/stories/pagination-server.stories.tsx:1]: Dont need a separate component entry here. Just need 1 sample use-case story inside `pagination.stories.tsx`

**Review by irfancoder — CHANGES_REQUESTED** [review]

File: packages/react/src/components/button.tsx [hunk 1]
```diff
@@ -95,7 +95,7 @@ const button_cva = cva(
   },
 );
 
-interface ButtonProps
+export interface ButtonProps
```
Comment (amalasyrafs) [inline packages/react/src/components/button.tsx:98]: exporting this ButtonProps so that could reuse the type across the library

**Comment by irfancoder** [comment]
@amalasyrafs check the reviewed `asChild` implementation & the expected usage in the story. Something to keep it in mind when building future components. Keep up the good work!

**Pull request merged**